local-volume/provisioner/pkg/deleter/deleter.go

```diff
@@ -17,28 +17,35 @@ limitations under the License.
 package deleter
 
 import (
+	"fmt"
 	"path/filepath"
+	"time"
 
 	"github.com/golang/glog"
-	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/types"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/common"
 
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/pkg/api/v1"
 )
 
-type Deleter interface {
-	// Cleanup and delete all its owned PVs that have been released
-	DeletePVs()
-}
+const (
+	pollInterval = 1 * time.Second
+	// TODO: is this too fast?
```

Comment:
The problem sounds similar to creating the volume twice.  Since controller will actively sync the state, we'll eventually reach the consistent point, so waiting doesn't seem to be strictly required?